content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: Forms
+---
+
+![forms example screenshot]()
+<!--
+- login
+- user settings
+- create repo
+-->
+
+Forms are used to complete tasks that require data input from the user. For example: creating a new repo, configuring settings, and logging in.
+
+Primer's form design guidelines aim to minimize the effort and cognitive load required to complete a task that involves a form.
+
+Note: this document is an early work in progress. More information will be added as our designers define how forms should work at GitHub. For example: form submission/saving patterns, form layouts, different kinds of forms, and much more
+<!--
+TODO:
+The following topics will be critical to defining strong guidelines for designing forms:
+- Form layout
+- Form structure and hierarchy
+- Validation messages that apply to the entire form
+- Form submission patterns
+- Patterns for grouping related fields
+- How to deal with forms that are very long or contain multiple parts
+- Fields that use a "progressive disclosure" pattern based on whether a setting has been toggled on
+-->
+
+## Form field anatomy
+
+Form fields are made up of the same parts, but they are laid out differently depending on the kind of input.
+
+![a diagram labeling the anatomy of a text input field]()
+![a diagram labeling the anatomy of a checkbox input field]()
+
+1. Label (required)
+2. Required field indicator
+3. Input (required)
+<!-- TODO: finish listing labeled parts -->
+
+### Label (required)
+Fields must be labeled so users understand what the field is used for.
+
+Labels should be descriptive and as short as possible: ideally, it will always fit on one line of text. Labels should be written in sentence-case. For example: "Repository name".
+
+If you're having trouble keeping label text short, consider using a [caption](#caption) to provide more context.
+
+In rare cases, a label may be hidden when the user has enough context to understand what the field does. When a label is visually hidden, label text must still be provided for assistive technology such as screen readers. For example: a search field with an icon that is placed in the spot a user would expect to find a search field may have a visually hidden label that reads "Search"
+
+### Required field indicator
+When a field is required to have a value, it should be visibly marked as required. For more information, see the [required fields](#required-fields) section.
+
+### Input (required)
+The input is what the user interacts with to set the value of the field.
+
+For more specific information about each input, visit the respective component documentation page. For example: [CheckboxInputField](), [TextInputField]()
+
+### Placeholder text
```

Comment:
@mperrotti I mentioned this in the TextInput component review, [but I proposed we completely do away with Placeholder text](https://github.com/github/primer/issues/538#:~:text=a%20loading%20state-,Placeholder%20text,-%F0%9F%94%97). [@jscholes had a similar suggestion](https://github.com/github/accessibility/issues/892#issuecomment-1077764961) as well. 
- https://github.com/github/primer/issues/538#:~:text=a%20loading%20state-,Placeholder%20text,-%F0%9F%94%97

Reply:
Thanks for the info. I'm going to remove the "Do / Don't" example so we're not explicitly encouraging the use of `placeholder`.